pkg/config/configobserver/configobserver.go

```diff
@@ -150,21 +159,23 @@ func (c *Controller) updateToken(ctx context.Context) error {
 
 // Updates the stored configs from the secrets in the cluster. (if present)
 func (c *Controller) updateConfig(ctx context.Context) error {
-	var nextConfig config.Controller
 	klog.V(2).Infof("Refreshing configuration from cluster secret")
 	secret, err := c.fetchSecret(ctx, "support")
 	if err != nil {
 		return err
 	}
 
-	if secret != nil {
-		nextConfig, err = LoadConfigFromSecret(secret)
+	c.supportSecret = secret
+	if secret == nil {
+		c.setSecretConfig(nil)
```

Comment:
Is this necessary? I think it will be `nil` anyway.

Reply:
I think I needed it to reset the secret config in case the support secret was deleted

Reply:
Ohh now I see it. We've done it before as well, but not in such obvious way